proposals/utils/pdf_diff_logic.py

```diff
@@ -0,0 +1,1132 @@
+
+    section_title = _('Data Management Plan')
+    
+    row_fields = ['dmp_file']
+
+class EmbargoSection(PDFSection):
+    '''Gets passed a proposal object'''
+
+    section_title = _('Aanmelding versturen')
+
+    row_fields = [
+        'embargo',
+        'embargo_end_date'
+    ]
+
+    def get_row_fields(self, obj):
+        rows = copy(self.row_fields)
+
+        if not obj.embargo:
+            rows.remove('embargo_end_date')
+
+        return rows
+    
+def get_extra_documents(object):
+    '''A function to retrieve all extra documents for a specific proposal.'''
+    from studies.models import Documents
+
+    extra_documents = []
+
+    for document in Documents.objects.filter(
+        proposal = object,
+        study__isnull = True
+    ):
+        extra_documents.append(document)
+    
+    return extra_documents
+    
+def create_context_pdf(context, model):
+    '''A function to create the context for the PDF.'''
+
+    context['general'] = GeneralSection(model)
+    context['wmo'] = WMOSection(model.wmo)
+
+    if model.wmo.status != model.wmo.NO_WMO:
+        context['metc'] = METCSection(model.wmo)
+    
+    context['trajectories'] = TrajectoriesSection(model)
+
+    if model.wmo.status == model.wmo.NO_WMO:
+
+        context['studies'] = []
+
+        for study in model.study_set.all():
+            study_sections = []
+            study_sections.append(StudySection(study))
+            if study.has_intervention:
+                study_sections.append(InterventionSection(study.intervention))
+            if study.has_observation:
+                study_sections.append(ObservationSection(study.observation))
+            if study.has_sessions:
+                study_sections.append(SessionsOverviewSection(study))
+                for session in study.session_set.all():
+                    study_sections.append(SessionSection(session))
+                    for task in session.task_set.all():
+                        study_sections.append(TaskSection(task))
+                study_sections.append(TasksOverviewSection(session))
+            study_sections.append(StudyOverviewSection(study))
+            study_sections.append(InformedConsentFormsSection(study.documents))
+            context['studies'].append(study_sections)
+
+        extra_documents = get_extra_documents(model)
+
+        extra_documents_objects = [ExtraDocumentsSection(document, count) for count, document \
+                                   in enumerate(extra_documents)]
+
+        if extra_documents_objects:
+            context['extra_documents'] = extra_documents_objects
+
+        if model.dmp_file:
+            context['dmp_file'] = DMPFileSection(model)
+            
+        context['embargo'] = EmbargoSection(model)
+    
+    return context
+
+def create_context_diff(context, models):
+    '''A function to create the context for the diff page.'''
+
+    p_proposal = models[0]
+    proposal = models[1]
```

Comment:
Why use a tuple to pass the two proposals? I think having them as separate arguments to the functions is a bit clearer